# Technical guidelines on creating MAGE-TAB files for Single Cell Expression Atlas

This document explains the way IDF and SDRF are constructed so that the experiment can be processed for Single Cell Expression Atlas, 
i.e. what fields are used by the processing pipeline.

This does NOT list all available fields and their controlled vocabulary. Please refer to the [single-cell curation guide](single_cell_curation_guide.md)
to see a list of all fields and how they should be curated.

The single-cell data analysis pipeline reads in IDF and SDRF files uploaded to the GitLab scxa-metadata repository 
to retrieve information about the runs and files to process. The following rules guarantee compliance of MAGE-TAB files with the 
[code for parsing IDF/SDRF and generation of the single cell config file](https://github.com/ebi-gene-expression-group/scxa-control-workflow/blob/master/bin/sdrfToNfConf.R).

## All Single-cell experiments

### IDF

All single-cell experiments need to have the following fields in their IDF:

|IDF field name|Used|Mandatory|Description|Example value|
|--- |--- |--- |--- |--- |
|Comment[EAExpectedClusters]|yes|no|How many clusters are expected in this data set? This is to set the range of pre-computed clusters (k values).|5|
|Comment[EAExperimentType]|no|yes|Type of Atlas analysis mode|[ baseline, differential ]|
|Experimental Factor Name|no|yes|The list of experimental factors in this experiment|[ cell type, developmental stage, ... ]|
|Protocol Name|no|yes|The list of protocol accessions in this experiment|[ P-MTAB-76362, ... ]|

<br>

## Smart-seq2 and other Smart-like experiments
### SDRF

The pipeline for data generated by Smart-seq2 and similar protocols expects the following fields in the SDRF:

|SDRF field|Used|Mandatory|Description|Example value|
|--- |--- |--- |--- |--- |
|Characteristics[organism]|yes|yes|Species Latin name for sample|[ Homo sapiens ... ]|
|Comment[ENA_RUN] OR Comment[RUN]|yes|yes|Run accession|[ ERR123456 ... ]|
|Comment[FASTQ_URI]|yes|yes|Path to FASTQ file (or just fastq file name for HCA smart-seq imports - the file names need to have _1 / _2 discriminitor to indicate endedness)|[ ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR123/006/ERR123456/ERR123456_1.fastq.gz ... ]|
|Comment[library construction]|yes|yes|Type of single-cell protocol|[ smart-seq2, smart-like ... ]|
|Comment[library_layout]|yes|yes|Single-end or paired-end run|[ SINGLE, PAIRED ]|
|Comment[library_strand]|yes|no|Strandedness of library|[ first strand, second strand, not applicable ]|
|Comment[single cell quality]|yes|yes|Determines whether run should be analysed or not|[ OK, not OK, OK, filtered ... ]|
|Comment[spike in]|yes|no|Name of the spike in set|[ ERCC ... ]|
|Comment[technical replicate group]|yes|no|Which runs belong to the same biological replicate|[ SAMEA123456 ... ]|
|Source Name|?|yes|Sample name|[ Sample 1 ... ]|

Note: Casing and spaces around square brackets are not important ("Comment[RUN]" == "comment [run]"). 
Also "Characteristics" and "Comment" are treated the same way ("Comment[single cell quality]" == "Characteristics[single cell quality]").

<br>

## 10x and other droplet-based experiments

### SDRF

The pipeline for data generated by 10x and similar protocols expects the following fields in the SDRF:

|SDRF field|Used|Mandatory|Description|Example value|
|--- |--- |--- |--- |--- |
|Source Name|?|yes|Sample name|[ Sample 1 ... ]|
|Characteristics[organism]|yes|yes|Species Latin name for sample|[ Homo sapiens ... ]|
|Comment[single cell quality]|yes|no|Determines whether run should be analysed or not|[ OK, not OK, OK, filtered ... ]|
|Comment[library construction]|yes|yes|Type of single-cell protocol|[ smart-seq2, smart-like ... ]|
|Comment[spike in]|yes|no|Name of the spike in set|[ ERCC ... ]|
|Comment[library_layout]|yes|no|Single-end or paired-end run|[ SINGLE, PAIRED ]|
|Comment[library_strand]|?|yes|Strandedness of library|[ first strand, second strand, not applicable ]|
|Comment[technical replicate group]|yes|no|Which runs belong to the same biological replicate|[ SAMEA123456 ... ]|
|Comment[ENA_RUN] OR Comment[RUN]|yes|yes|Run accession|[ ERR123456 ... ]|
|Comment[FASTQ_URI] (repeat x times)|yes|yes<sup>ยง</sup>|Path to FASTQ file (x = number of fastq files)|[ ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/MTAB/E-MTAB-6967/22089_1_AAACGGCG_S1_L001_R1_001.fastq.gz ]|
|Comment[SRA_URI] | yes | yes<sup>ยง</sup> | Path to SRA file | [ftp://ftp.sra.ebi.ac.uk/vol1/srr/SRR100/060/SRR10009460	]|
|Comment[read1 file]|yes|yes*|Name of the file containing sequencing read 1|[ 22089_1_AAACGGCG_S1_L001_R1_001.fastq.gz ]|
|Comment[read2 file]|yes|yes*|Name of the file containing sequencing read 2|[ 22089_1_AAACGGCG_S1_L001_R3_001.fastq.gz ]|
|Comment[index1 file]|yes|no|Name of the file containing the index1 read|[ 22089_1_AAACGGCG_S1_L001_R2_001.fastq.gz ]|
|Comment[umi barcode read]|?|yes*|||
|Comment[cell barcode read]|?|yes*|||
|Comment[cdna read]|?|yes*|||
|Comment[cell count]|yes|no|Number of cells detected in the library after any filtering steps|[ 1090 ... ]|
|Comment[cell count pre filtering]|yes|no|Number of cells detected in the library before quality filtering|[ 194 ... ]|
|Comment[HCA bundle uuid]|yes|yes*|For HCA experiments to point to location where to download fastq files||
|Comment[HCA bundle version]|yes|yes*|For HCA experiments to point to location where to download fastq files||
|Scan Name|yes+|yes|FASTQ file name (or run alias for droplet data)||

\* mandatory if applicable, otherwise not mandatory <br>
\+ used by the Atlas web app, not the processing pipeline <br>
ยง any of `FASTQ_URI` or `SRA_URI` is mandatory

Note that for droplet data, each row in the SDRF represents one run (i.e. one brokered BAM file). 
There must not be any duplication of the run due to more than one FASTQ file per run (as in paired-end bulk and plate-based single-cell sequencing).
To reference the raw FASTQ files that belong to each run, we use multiple Comment[FASTQ_URI] columns. 
The Scan Name column should contain a unique run alias, not the fastq file name.

